pilot/helpers/cli.py

```diff
@@ -448,9 +448,9 @@ def run_command_until_success(convo, command,
                     'command_id': command_id,
                     'success_message': success_message,
                 })
-            except TooDeepRecursionError as e:
+            except TooDeepRecursionError:
                 # this is only to put appropriate message in the response after TooDeepRecursionError is raised
-                raise TooDeepRecursionError(cli_response) if return_cli_response else e
+                raise TooDeepRecursionError(cli_response)
```

Comment:
we cant remove this if statement. This is how we get out of recursion

Reply:
The original one like:
```python 
if return_cli_response :
    raise TooDeepRecursionError(cli_response)
else:
    raise e
```

In both way, `TooDeepRecursionError` will be raised but just the text differ, if i am right.
So i move the text to
```python
class TooDeepRecursionError(Exception):
    def __init__(self, context=None, message='Recursion is too deep!'):
        if context:
            message = f"{message} (Context: {context})"
        super().__init__(message)
```
If no context is provided, will show the messgae `Recursion is too deep`.

Reply:
I will have to test this today or if you can share logs/screenshots of testing that would be great